### PR TITLE
CAMEL-20996: revive camel-solr and upgrade to 9.6.1

### PR DESCRIPTION
**Pull request by vdomrachev**
The goal of PR is to restore camel-solr component and update Solr to current 9.6.1 version.

### CODE REVIEW

File: components/camel-solr/pom.xml [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+            <artifactId>camel-support</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.camel</groupId>
+            <artifactId>camel-attachments</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.eclipse.angus</groupId>
+            <artifactId>angus-mail</artifactId>
+            <version>2.0.3</version>
```
Comment (oscerd) [inline components/camel-solr/pom.xml:49]: This should come from the parent pom property placeholder

File: components/camel-solr/pom.xml [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+        <version>4.8.0-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>camel-solr</artifactId>
+    <packaging>jar</packaging>
+    <name>Camel :: Solr</name>
+    <description>Camel Solr Support</description>
+
+    <properties>
+        <java.awt.headless>true</java.awt.headless>
```
Comment (oscerd) [inline components/camel-solr/pom.xml:34]: This should not be declared

File: components/camel-solr/pom.xml [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+        <dependency>
+            <groupId>org.eclipse.jetty</groupId>
+            <artifactId>jetty-server</artifactId>
+            <version>10.0.20</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.eclipse.jetty</groupId>
+            <artifactId>jetty-servlet</artifactId>
+            <version>10.0.20</version>
```
Comment (oscerd) [inline components/camel-solr/pom.xml:136]: From parent pom

File: components/camel-solr/pom.xml [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+    		<groupId>org.apache.solr</groupId>
+    		<artifactId>solr-extraction</artifactId>
+            <version>${solr-version}</version>
+            <scope>test</scope>    
+		</dependency>
+
+        <dependency>
+            <groupId>org.eclipse.jetty</groupId>
+            <artifactId>jetty-server</artifactId>
+            <version>10.0.20</version>
```
Comment (oscerd) [inline components/camel-solr/pom.xml:130]: From parent pom

File: components/camel-solr/src/main/docs/solr-component.adoc [hunk 5]
```diff
@@ -0,0 +1,170 @@
+= Solr Component
+:doctitle: Solr
+:shortname: solr
+:artifactid: camel-solr
+:description: Perform operations against Apache Lucene Solr.
+:since: 2.9
```
Comment (oscerd) [inline components/camel-solr/src/main/docs/solr-component.adoc:6]: This is a new component so the first version should be 4.8.0 and not 2.9

File: components/camel-solr/src/main/java/org/apache/camel/component/solr/SolrClientHandlerCloud.java [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    protected SolrClient getSolrClient() {
+        Optional<String> zkChrootOptional = Optional.ofNullable(solrConfiguration.getZkChroot());
+        List<String> urlList = getUrlListFrom(solrConfiguration);
+        CloudHttp2SolrClient.Builder builder = new CloudHttp2SolrClient.Builder(
+                urlList,
+                zkChrootOptional);
+        if (solrConfiguration.getCollection() != null && !solrConfiguration.getCollection().isEmpty()) {
+            builder.withDefaultCollection(solrConfiguration.getCollection());
+        }
+        //        if (solrConfiguration.getConnectionTimeout() != null) {
```
Comment (oscerd) [inline components/camel-solr/src/main/java/org/apache/camel/component/solr/SolrClientHandlerCloud.java:40]: Is it possible to set the connection Timeout?

Comment (oscerd) [reply components/camel-solr/src/main/java/org/apache/camel/component/solr/SolrClientHandlerCloud.java:40]: It should be possible if we declare the param in the configuration

File: components/camel-solr/src/test/java/org/apache/camel/component/solr/JettySolrFactory.java [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+            @Override
+            public X509Certificate[] getAcceptedIssuers() {
+                return null;
+            }
+        } };
+        final SSLContext sslContext = SSLContext.getInstance("TLS");
+        sslContext.init(null, trustAllCerts, new java.security.SecureRandom());
+        SSLContext.setDefault(sslContext);
+
+        // // Install the all-trusting trust manager
```
Comment (oscerd) [inline components/camel-solr/src/test/java/org/apache/camel/component/solr/JettySolrFactory.java:84]: Do we need this comments?

**Review by oscerd — CHANGES_REQUESTED** [review]
Just minor findings

File: components/camel-solr/pom.xml (hunk 2 above)
Comment (vdomrachev) [reply components/camel-solr/pom.xml]: fixed

File: components/camel-solr/pom.xml (hunk 3 above)
Comment (vdomrachev) [reply components/camel-solr/pom.xml:132]: Solr internally uses jetty-server 10.0.20 for testing purposes. 
solr/versions.lock
---
org.eclipse.jetty:jetty-server:10.0.22 (7 constraints: d86da720)
org.eclipse.jetty:jetty-servlet:10.0.22 (2 constraints: 3213f360)
org.eclipse.jetty:jetty-servlets:10.0.22 (1 constraints: 67059040)
---

If i upate to parent pom version than tests are failing with errors.

[INFO] Running org.apache.camel.component.solr.[1mInitSolrEndpointTest[m
[INFO] [1;32mTests run: [0;1;32m16[m, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.849 s -- in org.apache.camel.component.solr.[1mInitSolrEndpointTest[m
[INFO] Running org.apache.camel.component.solr.[1mSolrAddBeanTest[m
[ERROR] [1;31mTests [0;1mrun: [0;1m1[m, Failures: 0, [1;31mErrors: [0;1;31m1[m, Skipped: 0, Time elapsed: 0.116 s[1;31m <<< FAILURE![m -- in org.apache.camel.component.solr.[1mSolrAddBeanTest[m
[ERROR] org.apache.camel.component.solr.SolrAddBeanTest -- Time elapsed: 0.116 s <<< ERROR!
java.lang.NoClassDefFoundError: org/eclipse/jetty/server/SessionIdManager
	at org.apache.camel.component.solr.JettySolrFactory.createJetty(JettySolrFactory.java:112)
	at org.apache.camel.component.solr.JettySolrFactory.createJettyTestFixture(JettySolrFactory.java:131)
	at org.apache.camel.component.solr.SolrFixtures.createSolrFixtures(SolrFixtures.java:94)
	at org.apache.camel.component.solr.SolrComponentTestSupport.beforeClass(SolrComponentTestSupport.java:93)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
Caused by: java.lang.ClassNotFoundException: org.eclipse.jetty.server.SessionIdManager
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:525)
	... 6 more

These dependencies are used only for internal component testing. Mey be we can stay on these? And wait for solr to update or maybe in future replace with testcontainers? For the moment solr developers still use this org.apache.solr.embedded.JettySolrRunner.JettySolrRunner (10.0.20).

File: components/camel-solr/src/main/docs/solr-component.adoc (hunk 5 above)
Comment (vdomrachev) [reply components/camel-solr/src/main/docs/solr-component.adoc:6]: fixed

File: components/camel-solr/src/main/java/org/apache/camel/component/solr/SolrClientHandlerCloud.java (hunk 6 above)
Comment (vdomrachev) [reply components/camel-solr/src/main/java/org/apache/camel/component/solr/SolrClientHandlerCloud.java]: fixed.

File: components/camel-solr/src/test/java/org/apache/camel/component/solr/JettySolrFactory.java (hunk 7 above)
Comment (vdomrachev) [reply components/camel-solr/src/test/java/org/apache/camel/component/solr/JettySolrFactory.java]: removed commented code

File: components/camel-solr/pom.xml (hunk 3 above)
Comment (oscerd) [reply components/camel-solr/pom.xml:132]: Maybe add a property in the parent pom specific for this dependency

Comment (vdomrachev) [reply components/camel-solr/pom.xml]: moved version to camel/parent/pom property "jetty-for-solr-version "

**Review by oscerd — APPROVED** [review]

**Comment by oscerd** [comment]
LGTM and we can eventually re-iterate. Let's wait for @orpiske and other to review.

**Comment by vdomrachev** [comment]
Should i do something with failing checks? (PR Build (Camel 4) / build (17, false) (pull_request) Failing after 10m)

**Comment by vdomrachev** [comment]
Ok.Started full build. Should i commit all generated and not gitignored changed files? And should i remove added components/camel-solr/.gitignore file with "src/generated" and then commit all generated files from src/generated folder in this PR?

**Comment by oscerd** [comment]
Looks good to me.

File: components/camel-solr/src/main/java/org/apache/camel/component/solr/SolrClientHandlerCloud.java [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+        super(solrConfiguration);
+    }
+
+    protected SolrClient getSolrClient() {
+        Optional<String> zkChrootOptional = Optional.ofNullable(solrConfiguration.getZkChroot());
+        List<String> urlList = getUrlListFrom(solrConfiguration);
+        CloudHttp2SolrClient.Builder builder = new CloudHttp2SolrClient.Builder(
+                urlList,
+                zkChrootOptional);
+        if (solrConfiguration.getCollection() != null && !solrConfiguration.getCollection().isEmpty()) {
```
Comment (orpiske) [inline components/camel-solr/src/main/java/org/apache/camel/component/solr/SolrClientHandlerCloud.java:39]: Preferably, use [`ObjectHelper.isEmpty`](https://github.com/apache/camel/blob/main/core/camel-util/src/main/java/org/apache/camel/util/ObjectHelper.java#L210) / or ObjectHelper.notEmpty

File: components/camel-solr/src/main/java/org/apache/camel/component/solr/SolrClientHandlerConcurrentUpdate.java [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+        if (solrConfiguration.getConnectionTimeout() != null) {
+            builder.withConnectionTimeout(solrConfiguration.getConnectionTimeout(), TimeUnit.MILLISECONDS);
+        }
+        if (solrConfiguration.getIdleTimeout() != null) {
+            builder.withIdleTimeout(solrConfiguration.getIdleTimeout(), TimeUnit.MILLISECONDS);
+        }
+        if (solrConfiguration.getFollowRedirects() != null) {
+            builder.withFollowRedirects(solrConfiguration.getFollowRedirects());
+        }
+        Http2SolrClient httpSolrClient = builder.build();
```
Comment (orpiske) [inline components/camel-solr/src/main/java/org/apache/camel/component/solr/SolrClientHandlerConcurrentUpdate.java:42]: (non-binding suggestion) Maybe generalize this part here and use it in the `SolrClientHandlerHttp` class below, and other locations where you do the same thing?

File: components/camel-solr/src/main/java/org/apache/camel/component/solr/SolrConfiguration.java [hunk 10]
```diff
@@ -0,0 +1,389 @@
...
+
+    public SolrConfiguration deepCopy() {
+        try {
+            return (SolrConfiguration) this.clone();
+        } catch (CloneNotSupportedException e) {
+            LOG.error(
+                    String.format(
+                            "Could not generate new configuration based on configuration of existing endpoint %s",
+                            getSolrEndpoint()),
+                    e);
```
Comment (orpiske) [inline components/camel-solr/src/main/java/org/apache/camel/component/solr/SolrConfiguration.java:351]: I think there's no need to call `String.format` here. Calling just the `LOG.error` should be just fine.

File: components/camel-solr/src/main/java/org/apache/camel/component/solr/SolrProducer.java [hunk 11]
```diff
@@ -0,0 +1,302 @@
...
+        super(endpoint);
+    }
+
+    @Override
+    public void process(Exchange exchange) throws Exception {
+
+        String operation = (String) exchange.getIn().getHeader(SolrConstants.OPERATION);
+        if (operation == null) {
+            throw new IllegalArgumentException(SolrConstants.OPERATION + " header is missing");
+        }
```
Comment (orpiske) [inline components/camel-solr/src/main/java/org/apache/camel/component/solr/SolrProducer.java:60]: Please use [`NoSuchHeaderException`](https://github.com/apache/camel/blob/main/core/camel-api/src/main/java/org/apache/camel/NoSuchHeaderException.java) instead of `IllegalArgumentException`

File: components/camel-solr/src/main/java/org/apache/camel/component/solr/SolrProducer.java [hunk 12]
```diff
@@ -0,0 +1,302 @@
...
+        ModifiableSolrParams solrParams = new ModifiableSolrParams();
+        for (Map.Entry<String, Object> entry : exchange.getIn().getHeaders().entrySet()) {
+            if (entry.getKey().startsWith(SolrConstants.PARAM)) {
+                String paramName = entry.getKey().substring(SolrConstants.PARAM.length());
+                solrParams.add(paramName, entry.getValue().toString());
+            }
+        }
+
+        // solr collection
+        String solrCollection = exchange.getIn().getHeader(SolrConstants.COLLECTION, String.class);
```
Comment (orpiske) [inline components/camel-solr/src/main/java/org/apache/camel/component/solr/SolrProducer.java:81]: I'm under the impression this is a mandatory header, no? Probably we should throw `NoSuchHeaderException` if this is missing as well.

File: components/camel-solr/src/main/java/org/apache/camel/component/solr/SolrProducer.java [hunk 13]
```diff
@@ -0,0 +1,302 @@
...
+
+    private void query(
+            Exchange exchange, SolrClient solrClient, SolrConfiguration solrConfiguration, ModifiableSolrParams solrParams)
+            throws SolrServerException, IOException {
+        String solrCollection = exchange.getIn().getHeader(SolrConstants.COLLECTION, String.class);
+        SolrQuery solrQuery = new SolrQuery();
+        if (ObjectHelper.isNotEmpty(exchange.getMessage().getHeader(SolrConstants.QUERY_STRING))) {
+            solrQuery.setQuery(exchange.getMessage().getHeader(SolrConstants.QUERY_STRING, String.class));
+        } else {
+            throw new IllegalArgumentException("Query String needs to be set as header while querying Solr");
```
Comment (orpiske) [inline components/camel-solr/src/main/java/org/apache/camel/component/solr/SolrProducer.java:183]: `NoSuchHeaderException` here too (also in other places where we need to check for mandatory headers).

File: components/camel-solr/src/test/java/org/apache/camel/component/solr/SolrCloudFixture.java [hunk 14]
```diff
@@ -0,0 +1,179 @@
...
+        String zkAddr = miniCluster.getZkServer().getZkAddress();
+        String zkHost = miniCluster.getZkServer().getZkHost();
+
+        buildZooKeeper(zkHost, zkAddr, new File(solrHome), "solrconfig.xml", "schema.xml");
+        List<JettySolrRunner> jettys = miniCluster.getJettySolrRunners();
+        for (JettySolrRunner jetty : jettys) {
+            if (!jetty.isRunning()) {
+                LOG.warn("JETTY NOT RUNNING!");
+            } else {
+                LOG.info("JETTY RUNNING AT " + jetty.getBaseUrl() + " PORT " + jetty.getLocalPort());
```
Comment (orpiske) [inline components/camel-solr/src/test/java/org/apache/camel/component/solr/SolrCloudFixture.java:87]: Use log placeholders, please. `LOG.info("JETTY RUNNING AT {}:{}", jetty.getBaseUrl(), jetty.getLocalPort());`

File: components/camel-solr/src/test/java/org/apache/camel/component/solr/SolrCloudFixture.java [hunk 15]
```diff
@@ -0,0 +1,179 @@
...
+        return server.request(request);
+    }
+
+    public static void putConfig(
+            String confName, SolrZkClient zkClient, File solrhome, final String srcName,
+            String destName)
+            throws Exception {
+        File file = new File(solrhome, "collection1" + File.separator + "conf" + File.separator + srcName);
+        if (!file.exists()) {
+            LOG.info("zk skipping " + file.getAbsolutePath() + " because it doesn't exist");
```
Comment (orpiske) [inline components/camel-solr/src/test/java/org/apache/camel/component/solr/SolrCloudFixture.java:133]: Log placeholders here and in all other places.

File: components/camel-solr/src/test/java/org/apache/camel/component/solr/SolrUpdateTest.java [hunk 16]
```diff
@@ -0,0 +1,370 @@
...
+    public void init() {
+        solrEndpoint = getMandatoryEndpoint(solrRouteUri(), SolrEndpoint.class);
+    }
+
+    //	@Override
+    //    @BeforeEach
+    //    public void setUp() {
+    //        super.setUp();
+    //        solrEndpoint = getMandatoryEndpoint(solrRouteUri(), SolrEndpoint.class);
+    //    }
```
Comment (orpiske) [inline components/camel-solr/src/test/java/org/apache/camel/component/solr/SolrUpdateTest.java:56]: This commented code should be removed.

File: components/camel-solr/src/test/resources/solr/collection1/conf/solrconfig.xml [hunk 17]
```diff
@@ -0,0 +1,1818 @@
...
+    <!--
+       <lst name="spellchecker">
+         <str name="name">jarowinkler</str>
+         <str name="field">spell</str>
+         <str name="classname">solr.DirectSolrSpellChecker</str>
+         <str name="distanceMeasure">
+           org.apache.lucene.search.spell.JaroWinklerDistance
+         </str>
+       </lst>
+     -->
```
Comment (orpiske) [inline components/camel-solr/src/test/resources/solr/collection1/conf/solrconfig.xml:1280]: Do we need these commented bits?

File: test-infra/camel-test-infra-solr/src/test/java/org/apache/camel/test/infra/solr/services/SolrLocalContainerService.java [hunk 18]
```diff
@@ -0,0 +1,94 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.output.Slf4jLogConsumer;
+import org.testcontainers.containers.wait.strategy.Wait;
+
+public class SolrLocalContainerService implements SolrService, ContainerService<GenericContainer> {
+    public static final String CONTAINER_IMAGE = "solr:9.6.1";
+    public static final String CONTAINER_NAME = "solr";
+    private static final int PORT = 8983;
```
Comment (orpiske) [inline test-infra/camel-test-infra-solr/src/test/java/org/apache/camel/test/infra/solr/services/SolrLocalContainerService.java:30]: This needs to be in the `container.properties` file, so we can automate the check for container updates.

**Review by orpiske — CHANGES_REQUESTED** [review]
Thanks for your contribution! Overall I think it looks pretty good!

I did left a few comments because the code seems to be based on the previous `camel-solr` component which has not passed through the comprehensive cleanups we have been doing since 4.0. 

I think they are mostly small things that should be easy to do and that would help us ensure that the component is consistent w/ the rest of the code. 

On top of those, it also has not had the documentation reviewed for errors, typos and other mistakes (that we did around 4.3 - 4.4 timeframe) ... so, we'll need to create a ticket for that as well. 

TL;DR: good contribution ... just need to be adjusted to match the current code in a few areas.

File: components/camel-solr/src/main/java/org/apache/camel/component/solr/SolrProducer.java (hunk 12 above)
Comment (vdomrachev) [reply components/camel-solr/src/main/java/org/apache/camel/component/solr/SolrProducer.java:81]: org.apache.solr.client.solrj.impl.Http2SolrClient.request(SolrRequest<?>, String)
...
    if (ClientUtils.shouldApplyDefaultCollection(collection, solrRequest)) {
      collection = defaultCollection;
    }

There is defaultCollection in SolrClient. So its not mandatory in Exchange headers.

File: components/camel-solr/src/main/java/org/apache/camel/component/solr/SolrConfiguration.java (hunk 10 above)
Comment (vdomrachev) [reply components/camel-solr/src/main/java/org/apache/camel/component/solr/SolrConfiguration.java:351]: fixed

File: components/camel-solr/src/main/java/org/apache/camel/component/solr/SolrClientHandlerCloud.java (hunk 8 above)
Comment (vdomrachev) [reply components/camel-solr/src/main/java/org/apache/camel/component/solr/SolrClientHandlerCloud.java:39]: fixed

File: components/camel-solr/src/main/java/org/apache/camel/component/solr/SolrProducer.java (hunk 11 above)
Comment (vdomrachev) [reply components/camel-solr/src/main/java/org/apache/camel/component/solr/SolrProducer.java:60]: fixed

File: components/camel-solr/src/main/java/org/apache/camel/component/solr/SolrProducer.java (hunk 13 above)
Comment (vdomrachev) [reply components/camel-solr/src/main/java/org/apache/camel/component/solr/SolrProducer.java:183]: fixed

File: components/camel-solr/src/test/java/org/apache/camel/component/solr/SolrCloudFixture.java (hunk 14 above)
Comment (vdomrachev) [reply components/camel-solr/src/test/java/org/apache/camel/component/solr/SolrCloudFixture.java:87]: fixed

File: components/camel-solr/src/test/java/org/apache/camel/component/solr/SolrCloudFixture.java (hunk 15 above)
Comment (vdomrachev) [reply components/camel-solr/src/test/java/org/apache/camel/component/solr/SolrCloudFixture.java:133]: fixed

File: components/camel-solr/src/test/java/org/apache/camel/component/solr/SolrUpdateTest.java (hunk 16 above)
Comment (vdomrachev) [reply components/camel-solr/src/test/java/org/apache/camel/component/solr/SolrUpdateTest.java:56]: fixed

File: components/camel-solr/src/test/resources/solr/collection1/conf/solrconfig.xml (hunk 17 above)
Comment (vdomrachev) [reply components/camel-solr/src/test/resources/solr/collection1/conf/solrconfig.xml:1280]: fixed. Made minimal solrconfig.xml

File: test-infra/camel-test-infra-solr/src/test/java/org/apache/camel/test/infra/solr/services/SolrLocalContainerService.java (hunk 18 above)
Comment (vdomrachev) [reply test-infra/camel-test-infra-solr/src/test/java/org/apache/camel/test/infra/solr/services/SolrLocalContainerService.java]: fixed analogically to redis infra component

File: components/camel-solr/src/main/java/org/apache/camel/component/solr/SolrClientHandlerConcurrentUpdate.java (hunk 9 above)
Comment (vdomrachev) [reply components/camel-solr/src/main/java/org/apache/camel/component/solr/SolrClientHandlerConcurrentUpdate.java]: fixed

**Comment by vdomrachev** [comment]
Unfortunately, it looks like we have to wait for https://issues.apache.org/jira/browse/SOLR-17069 to make this PR work in camel 4.8.x . My working build of my project with updated camel-solr component was based on 4.4.x branch of camel, spring boot 3.2.5. 

However 4.8.x camel is linked with spring boot 3.3.1. I have tried to build and test my project with 4.8.0-SNAPSHOT camel-solr. There are runtime errors of incompatibility of jersey http client libs in this built, that i havent met on 4.4.x branch. 

Another way is to restore usage of deprecated apache http client and wait for solr to update. I've commited these changes.
I've built 4.8.0-SNAPSHOT version of camel-solr, camel-spring-boot with apache http client. Then used this in my app and tested. 
Now It works.

Anyway, work in this PR may be helpful for future migration when solr updates to jersey 12.x. 

Thanks for your attention and for your time.

**Comment by oscerd** [comment]
Thanks for this, I'm going to merge this, probably beginning of the next week.

**Comment by orpiske** [comment]
BTW, do we have a ticket for this? It may be beneficial to track this on a Jira ticket, so that it is listed on the release, trackable, etc ... if there's no ticket, I can create one.

**Comment by oscerd** [comment]
No ticket.

**Comment by vdomrachev** [comment]
May be you can reopen and update https://issues.apache.org/jira/browse/CAMEL-18577 ?

**Comment by orpiske** [comment]
> May be you can reopen and update https://issues.apache.org/jira/browse/CAMEL-18577 ?

No. That is already resolved. 

I created https://issues.apache.org/jira/browse/CAMEL-20996 for this.

**Comment by oscerd** [comment]
This is still not ready to be merged.

**Comment by vdomrachev** [comment]
Is it because of failing PR build checks and changes in generated files? I'm still not able to successfully run full repo build because of failing tests in not dependent to this PR modules. But clean install -Dquickly runs successfully. I can do some more tries to do full build

**Comment by oscerd** [comment]
I can try to merge and then generate later

**Comment by oscerd** [comment]
I'm starting to merge now and I regenerate the metadata.

**Comment by oscerd** [comment]
Merged and regen
